GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1534 @@
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+            return newGroup;
+
+            #endregion
+        }
+
+        private static async Task PrintAllRoles(IActiveDirectoryClient client, string searchString)
+        {
+            #region Get All Roles
+
+            //*********************************************************************
+            // Get All Roles
+            //*********************************************************************
+            List<IDirectoryRole> foundRoles = null;
+            try
+            {
+                IPagedCollection<IDirectoryRole> rolesCollection = await client.DirectoryRoles.ExecuteAsync();
+                foundRoles = rolesCollection.CurrentPage.ToList();
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError getting Roles {0} {1}", e.Message,
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+
+            if (foundRoles != null && foundRoles.Count > 0)
+            {
+                foreach (IDirectoryRole role in foundRoles)
+                {
+                    Console.WriteLine("\n Found Role: {0} {1} {2} ",
+                        role.DisplayName, role.Description, role.ObjectId);
+                }
+            }
+            else
+            {
+                Console.WriteLine("Role Not Found {0}", searchString);
+            }
+
+            #endregion
+        }
+
+        private static async Task PrintServicePrincipals(IActiveDirectoryClient client)
+        {
+            #region Get Service Principals
+
+            //*********************************************************************
+            // get the Service Principals
+            //*********************************************************************
+            IPagedCollection<IServicePrincipal> servicePrincipals = null;
+            try
+            {
+                servicePrincipals = await client.ServicePrincipals.ExecuteAsync();
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError getting Service Principal {0} {1}",
+                    e.Message, e.InnerException != null ? e.InnerException.Message : "");
+            }
+            if (servicePrincipals != null)
+            {
+                do
+                {
+                    List<IServicePrincipal> servicePrincipalsList = servicePrincipals.CurrentPage.ToList();
+                    foreach (IServicePrincipal servicePrincipal in servicePrincipalsList)
+                    {
+                        Console.WriteLine("Service Principal AppId: {0}  Name: {1}", servicePrincipal.AppId,
+                            servicePrincipal.DisplayName);
+                        // find the Graph API service principal objectId
+                        if (servicePrincipal.AppId == "00000002-0000-0000-c000-000000000000")
```

Comment:
moved.

---
In reply to: [80588237](https://github.com/Azure-Samples/active-directory-dotnet-graphapi-console/pull/37#discussion_r80588237) [](ancestors = 80588237)